components/forms/VestingForm.js

```diff
@@ -0,0 +1,163 @@
+  const [amount, setAmount] = useState("0");
+  const [unixEpochTime, setUnixEpochTime] = useState("0");
+  const [delayed, setDelayed] = useState(true);
+  const [memo, setMemo] = useState("");
+  const [gas, setGas] = useState(200000);
+  const [gasPrice, _setGasPrice] = useState(state.chain.gasPrice);
+  const [_processing, setProcessing] = useState(false);
+  const [addressError, setAddressError] = useState("");
+
+  const createTransaction = (txToAddress, txAmount, txGas, txTime) => {
+    const amountInAtomics = Decimal.fromUserInput(
+      txAmount,
+      Number(state.chain.displayDenomExponent),
+    ).atomics;
+    const msgCreateVestingAccount = {
+      fromAddress: props.address,
+      toAddress: txToAddress,
+      endTime: txTime,
+      amount: [
+        {
+          amount: amountInAtomics,
+          denom: state.chain.denom,
+        },
+      ],
+      delayed: delayed,
+    };
+    const msg = {
+      typeUrl: "/cosmos.vesting.v1beta1.MsgCreateVestingAccount",
+      value: msgCreateVestingAccount,
+    };
+    const fee = calculateFee(Number(txGas), gasPrice);
+    return {
+      accountNumber: props.accountOnChain.accountNumber,
+      sequence: props.accountOnChain.sequence,
+      chainId: state.chain.chainId,
+      msgs: [msg],
+      fee: fee,
+      memo: memo,
+    };
+  };
+
+  const handleCreate = async () => {
+    const toAddressError = checkAddress(toAddress, state.chain.addressPrefix);
+    if (toAddressError) {
+      setAddressError(`Invalid address for network ${state.chain.chainId}: ${toAddressError}`);
+      return;
+    }
+
+    setProcessing(true);
+    const tx = createTransaction(toAddress, amount, gas, unixEpochTime);
+    console.log(tx);
+    const dataJSON = JSON.stringify(tx);
+    const res = await axios.post("/api/transaction", { dataJSON });
+    const { transactionID } = res.data;
+    props.router.push(`${props.address}/transaction/${transactionID}`);
+  };
+
+  return (
+    <StackableContainer lessPadding>
+      <button className="remove" onClick={() => props.closeForm()}>
+        ✕
+      </button>
+      <h2>Create New Vesting Account</h2>
+      <div className="form-item">
+        <Input
+          label="Recieving Address"
+          name="toAddress"
+          value={toAddress}
+          onChange={(e) => setToAddress(e.target.value)}
+          error={addressError}
+          placeholder={`E.g. ${exampleAddress(0, state.chain.addressPrefix)}`}
+        />
+      </div>
+      <div className="form-item">
+        <Input
+          label={`Amount (${state.chain.displayDenom})`}
+          name="amount"
+          type="number"
+          value={amount}
+          onChange={(e) => setAmount(e.target.value)}
+        />
+      </div>
+      <div className="form-item">
+        <Input
+          label="Unix Epoch Time"
+          name="unixEpochTime"
+          type="number"
+          value={unixEpochTime}
+          onChange={(e) => setUnixEpochTime(e.target.value)}
+        />
```

Comment:
This naming is about the format, but does not help understand what the value is used for. Can we use something like "Vesting end time (UNIX timestamp)". Is this in seconds? Nanoseconds?